landoapi/cli.py

```diff
@@ -57,7 +57,9 @@ def init():
 @click.argument("celery_arguments", nargs=-1, type=click.UNPROCESSED)
 def worker(celery_arguments):
     """Initialize a Celery worker for this app."""
-    for system in get_subsystems(exclude=["repo_clone_subsystem"]):
+    from landoapi.app import repo_clone_subsystem
```

Comment:
I still think it's backwards to import all the things you *don't* need.

I notice there's a pattern in `cli.py` where we check the subsystems are ready and then call out to some other `landoapi` module. Have you considered moving the `ensure_ready` calls into these other modules? For example in `landing_worker` you could ensure the subsystems are ready in the `LandingWorker()` constructor.

I don't know if this is a big architecture change, so it may not be worth doing. Just seems like an anti-pattern to me. :)

Reply:
I agree 100% that this could benefit from a refactor -- in terms of `ready` and `ensure_ready`, and it's worth doing though it is a bit of a bigger change, so I'll have to do it in a different PR.

> I still think it's backwards to import all the things you don't need.

I also agree here. There are some alternatives that would clean this up, let me see what I can add in a different PR to help with this.